src/server/saved_objects/service/lib/repository.js

```diff
@@ -224,6 +224,35 @@ export class SavedObjectsRepository {
     );
   }
 
+  async deleteByQuery(options = {}) {
+    const {
+      type,
+      search,
+      searchFields,
+      namespace,
+    } = options;
+
+    if (searchFields && !Array.isArray(searchFields)) {
+      throw new TypeError('options.searchFields must be an array');
+    }
+
+    const esOptions = {
+      index: this._index,
+      ignore: [404],
+      body: {
+        version: true,
+        ...getSearchDsl(this._mappings, this._schema, {
+          namespace,
+          search,
+          searchFields,
+          type,
+        })
+      }
+    };
+
+    return await this._writeToCluster('deleteByQuery', esOptions);
```

Comment:
For our use case, it seems like we'd want to set `conflicts=proceed` mentioned here: https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-delete-by-query.html